rsocket-core/src/main/java/io/rsocket/DuplexConnection.java

```diff
@@ -79,6 +60,12 @@
    */
   Flux<ByteBuf> receive();
 
+  /**
+   * @param errorFrame
+   * @return
+   */
+  void terminate(ByteBuf frame, RSocketErrorException terminalError);
```

Comment:
It looks like the frame is always `ErrorFrameCodec.encode(duplexConnection.alloc(), 0, exception)`. Maybe this could become a one-argument method `terminate(RSocketErrorException)`.

Reply:
Removed Exception at all and left ByteBuf frame only